lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -1497,25 +1512,25 @@ fn ignore_unexpected_static_invoice() {
 	create_unannounced_chan_between_nodes_with_value(&nodes, 1, 2, 1_000_000, 0);
 
 	// Initiate payment to the sender's intended offer.
-	let dummy_blinded_path_to_always_online_node = BlindedMessagePath::from_raw(
-		nodes[1].node.get_our_node_id(), test_utils::pubkey(42),
-		vec![BlindedHop { blinded_node_id: test_utils::pubkey(42), encrypted_payload: vec![42; 32] }]
-	);
-	let (offer_builder, offer_nonce) = nodes[2].node.create_async_receive_offer_builder(vec![dummy_blinded_path_to_always_online_node.clone()]).unwrap();
+	let blinded_paths_to_always_online_node = nodes[1].message_router.create_blinded_paths(
+		nodes[1].node.get_our_node_id(),
+		MessageContext::Offers(OffersContext::InvoiceRequest { nonce: Nonce([42; 16]) }),
+		Vec::new(), &secp_ctx
+	).unwrap();
+	let (offer_builder, offer_nonce) = nodes[2].node.create_async_receive_offer_builder(blinded_paths_to_always_online_node.clone()).unwrap();
 	let offer = offer_builder.build().unwrap();
 	let amt_msat = 5000;
 	let payment_id = PaymentId([1; 32]);
-	// Set the random bytes so we can predict the offer outbound payment context nonce.
-	*nodes[0].keys_manager.override_random_bytes.lock().unwrap() = Some([42; 32]);
 	nodes[0].node.pay_for_offer(&offer, None, Some(amt_msat), None, payment_id, Retry::Attempts(0), None).unwrap();
 
 	// Don't forward the invreq since we don't support retrieving the static invoice from the
-	// recipient's LSP yet, instead just provide the invoice directly to the payer.
-	let _invreq_om = nodes[0].onion_messenger.next_onion_message_for_peer(nodes[1].node.get_our_node_id()).unwrap();
+	// recipient's LSP yet, instead manually construct the responses below.
+	let invreq_om = nodes[0].onion_messenger.next_onion_message_for_peer(nodes[1].node.get_our_node_id()).unwrap();
+	let invreq_reply_path = extract_invoice_request_reply_path(&nodes[1], &invreq_om);
 
 	// Create a static invoice with the same payment_id but corresponding to a different offer.
```

Comment:
IIUC, the static invoice doesn't have a payment id. Rather the reply path that it is sent over does.